File: coding-patterns/01-two-pointers/011.py
# Find Non-Duplicate Number Instances (Easy)

class Solution:
  def moveElements(self, arr):
    next_non_duplicate = 1

    i = 0
    while i < len(arr):
      if arr[next_non_duplicate - 1] != arr[i]:
        arr[next_non_duplicate] = arr[i]
        next_non_duplicate += 1
      i += 1

    return next_non_duplicate

File: coding-patterns/01-two-pointers/011.cpp
// Find Non-Duplicate Number Instances (Easy)

using namespace std;

#include <iostream>
#include <vector>

class Solution {
public:
  static int remove(vector<int> &arr) {
    int start = 0;
    int end = 1;
    int count = 0;
    while (start < arr.size()) {
      while (end < arr.size() && arr[end] == arr[start])
        end++;
      start = end;
      end++;
      count++;
    }
    return count;
  }
};